src/main/java/seedu/address/logic/commands/AddLinkCommand.java

```diff
@@ -1,82 +1,85 @@
-
-    private final Index index;
+    public static final String MESSAGE_USAGE = "[" + COMMAND_WORD + "]: Add link/s to a the module "
+            + "using its module code, link URL, and a user-defined alias.\n"
+            + "A 'm/' flag should be appended to the front the module code;\n"
+            + "a 'l/' flag should be appended to the front of each link URL;\n"
+            + "a 'la/' flag should be appended to the front of each link alias.\n"
+            + "Example: " + COMMAND_WORD + " " + PREFIX_MODULE_CODE + "GEA1000 "
+            + PREFIX_MODULE_LINK_ALIAS + "coursemo " + PREFIX_MODULE_LINK_URL + "coursemology.org";
+
+    public static final String MESSAGE_ADD_LINK_SUCCESS = "Successfully added link/s to module code [%1$s]!";
+    public static final String MESSAGE_NOT_EDITED = "At least one link url and alias must be added.";
+    public static final String MESSAGE_DUPLICATE_LINK_ALIAS = "The link alias [%1$s] already exists"
+            + " in the module with module code [%2$s].";
+    public static final String MESSAGE_DUPLICATE_LINK_URL = "The link URL [%1$s] already exists"
+            + " in the module with module code [%2$s] under the alias [%3$s].";
+
+    private final ModuleCode moduleCode;
     private final Set<Link> links;
 
     /**
      * Creates an AddLinkCommand for the addition of links to a module
-     * @param index index of the module based on the filtered module list
-     * @param links links to add to the module
+     * @param moduleCode module code of the module in which links will be added
+     * @param links links to add to the specified module
      */
-    public AddLinkCommand(Index index, Set<Link> links) {
-        requireAllNonNull(index, links);
-        this.index = index;
+    public AddLinkCommand(ModuleCode moduleCode, Set<Link> links) {
+        requireAllNonNull(moduleCode, links);
+        this.moduleCode = moduleCode;
         this.links = links;
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
-        ObservableList<Module> lastShownList = model.getFilteredModuleList();
-        if (index.getZeroBased() >= lastShownList.size()) {
-            throw new CommandException(Messages.MESSAGE_INVALID_MODULE_DISPLAYED_INDEX);
-        }
 
-        Module moduleToEdit = lastShownList.get(index.getZeroBased());
-        Module editedModule = createEditedModule(moduleToEdit, links);
-
-        if (moduleToEdit.getLinks().equals(editedModule.getLinks())) {
-            throw new CommandException(MESSAGE_DUPLICATE_LINK + index.getOneBased() + "]");
+        Module moduleToEdit = null;
+        try {
+            moduleToEdit =
+                    model.getModuleUsingModuleCode(moduleCode, true);
+        } catch (ModuleNotFoundException e) {
+            throw new CommandException(String.format(MESSAGE_NO_MODULE_IN_FILTERED_LIST,
```

Comment:
```suggestion
            throw new CommandException(String.format(Messages.MESSAGE_NO_MODULE_IN_FILTERED_LIST,
```
Inconsistent usage of `Messages.` and to adhere to group consensus on how `messages` should be used